File: Strings/pagina04/2253/2253.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin', 'utf-8')
const lines = input.split('\n')

const tem_maiuscula = (str) => /[A-Z]/.test(str)
const tem_minuscula = (str) => /[a-z]/.test(str)
const tem_numero = (str) => /[0-9]/.test(str)

while(lines.length){

    let curr = lines.shift()

    if(curr < 6 || curr > 32){
        console.log(`Senha invalida.`)
        continue
    }

    let maiuscula = false
    let minuscula = false
    let numero = false
    let errado = false
    for(let i = 0; i < curr.length; i++){
        let char = curr[i]
        if(tem_maiuscula(char)){
            maiuscula = true

        } else if (tem_minuscula(char)){
            minuscula = true

        } else if(tem_numero(char)){
            numero = true
        } else {
            errado = true
            break
        }
    }

    if(errado || !(maiuscula && minuscula && numero)){
        console.log(`Senha invalida.`)
    } else {
        console.log(`Senha valida.`)
    }
}

File: Strings/pagina04/2253/2253.cpp
#include <iostream>
#include <cctype>

using namespace std;

int main(){

	int n;
	bool maiuscula, minuscula, numero, errado;
	string str;

	while(getline(cin, str)){
		n = str.length();

		if(n < 6 || n > 32){
			cout << "Senha invalida." << endl;
			continue;
		}
		
		errado = false, maiuscula = false, minuscula = false, numero = false;
		for(char ch: str){
			if(isalpha(ch)){
				if(isupper(ch)) maiuscula = true;
				else minuscula = true;
				
			} else if(isdigit(ch)){
				numero = true;
			} else {
				errado = true;
				break;
			}
		}

		if(errado || !(maiuscula && minuscula && numero)){
			cout << "Senha invalida." << endl;
		} else {
			cout << "Senha valida." << endl;
		}
	}
	
	return 0;
}